WalletWasabi.Fluent/ViewModels/Login/LoginViewModel.cs

```diff
@@ -6,76 +6,54 @@
 using WalletWasabi.Fluent.ViewModels.Login.PasswordFinder;
 using WalletWasabi.Fluent.ViewModels.Navigation;
 using WalletWasabi.Fluent.ViewModels.Wallets;
+using WalletWasabi.Services;
 using WalletWasabi.Userfacing;
 
 namespace WalletWasabi.Fluent.ViewModels.Login
 {
 	[NavigationMetaData(Title = "Login")]
 	public partial class LoginViewModel : RoutableViewModel
 	{
-		private readonly WalletManagerViewModel _walletManagerViewModel;
-		private readonly ClosedWalletViewModel _closedWalletViewModel;
-
 		[AutoNotify] private string _password;
 		[AutoNotify] private bool _isPasswordIncorrect;
 		[AutoNotify] private bool _isPasswordNeeded;
 		[AutoNotify] private string _walletName;
 
 		public LoginViewModel(WalletManagerViewModel walletManagerViewModel, ClosedWalletViewModel closedWalletViewModel)
 		{
-			_walletManagerViewModel = walletManagerViewModel;
-			_closedWalletViewModel = closedWalletViewModel;
-
-			var wallet = _closedWalletViewModel.Wallet;
-			var legalChecker = walletManagerViewModel.LegalChecker;
-
+			var wallet = closedWalletViewModel.Wallet;
 			IsPasswordNeeded = !wallet.KeyManager.IsWatchOnly;
-
-			WalletIcon = wallet.KeyManager.Icon;
-			IsHardwareWallet = wallet.KeyManager.IsHardwareWallet;
 			_walletName = wallet.WalletName;
 			_password = "";
+			WalletIcon = wallet.KeyManager.Icon;
+			IsHardwareWallet = wallet.KeyManager.IsHardwareWallet;
 
 			NextCommand = ReactiveCommand.CreateFromTask(async () =>
 			{
-				IsPasswordIncorrect = await Task.Run(async () =>
-				{
-					if (!IsPasswordNeeded)
-					{
-						return false;
-					}
+				string? compatibilityPasswordUsed = null;
 
-					if (PasswordHelper.TryPassword(wallet.KeyManager, Password, out var compatibilityPasswordUsed))
-					{
-						if (compatibilityPasswordUsed is { })
-						{
-							await ShowErrorAsync(PasswordHelper.CompatibilityPasswordWarnMessage, "Compatibility password was used");
-						}
+				IsPasswordIncorrect = !await Task.Run(() => wallet.TryLogin(Password, out compatibilityPasswordUsed));
```

Comment:
Rename to IsPasswordCorrect so we are not constantly inverting the logic?

Reply:
fixed